dev/src/bulk-writer.ts

```diff
@@ -190,6 +195,15 @@ class BulkCommitBatch {
     return this.processOperation(documentRef);
   }
 
+  /**
+   * Adds the provided operation to the WriteBatch. Returns a promise that
+   * resolves with the result of the operation.
+   */
+  addOperation(operation: BulkWriterOperation): Promise<WriteResult> {
+    this.writeBatch.addOperation(operation.documentRef, operation._data);
```

Comment:
My dream of somehow removing the BulkCommitBatch is growing more and more vivid. Certainly not in this PR though :)